File: framec_tests/csharp/Naming/NamingTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using NUnit.Framework;

namespace Naming
{
    class NamingController : Naming
    {
        public NamingController() : base()
        {
        }
        public void snake_action_do(int snake_param)
        {
            this.snake_log.Add(snake_param);
        }

        public void CamelAction_do(int CamelParam)
        {
            this.CamelLog.Add(CamelParam);
        }

        public void action123_do(int param123)
        {
            this.log123.Add(param123);
        }

        public void logFinal_do(int r)
        {
            this.finalLog.Add(r);
        }
    }
    [TestFixture]
    public class NamingTest
    {
        [Test]
        public void TestFollowNamingWorks()
        {
            NamingController sm = new NamingController();

            sm.snake_event(1);
            Assert.AreEqual("1", sm.state_info());
            sm.snake_event(2);
            Assert.AreEqual("0", sm.state_info());
            sm.snake_event(1);
            Assert.AreEqual("1", sm.state_info());
            sm.CamelEvent(3);
            Assert.AreEqual("0", sm.state_info());
            sm.snake_event(1);
            Assert.AreEqual("1", sm.state_info());
            sm.event123(4);
            Assert.AreEqual("0", sm.state_info());
            Assert.AreEqual(new List<int> { 1103, 1104, 1105 }, sm.finalLog);
            sm.finalLog = new List<int>();

            sm.CamelEvent(11);
            Assert.AreEqual("2", sm.state_info());
            sm.snake_event(2);
            Assert.AreEqual("0", sm.state_info());
            sm.CamelEvent(11);
            Assert.AreEqual("2", sm.state_info());
            sm.CamelEvent(3);
            Assert.AreEqual("0", sm.state_info());
            sm.CamelEvent(11);
            Assert.AreEqual("2", sm.state_info());
            sm.event123(4);
            Assert.AreEqual("0", sm.state_info());
            Assert.AreEqual(new List<int> { 1213, 1214, 1215 }, sm.finalLog);
            sm.finalLog = new List<int>();

            sm.event123(21);
            Assert.AreEqual("3", sm.state_info());
            sm.snake_event(2);
            Assert.AreEqual("0", sm.state_info());
            sm.event123(21);
            Assert.AreEqual("3", sm.state_info());
            sm.CamelEvent(3);
            Assert.AreEqual("0", sm.state_info());
            sm.event123(21);
            Assert.AreEqual("3", sm.state_info());
            sm.event123(4);
            Assert.AreEqual("0", sm.state_info());
            Assert.AreEqual(new List<int> { 1323, 1324, 1325 }, sm.finalLog);

            Assert.AreEqual(new List<int> { 1103, 1213, 1323 }, sm.snake_log);
            Assert.AreEqual(new List<int> { 1104, 1214, 1324 }, sm.CamelLog);
            Assert.AreEqual(new List<int> { 1105, 1215, 1325 }, sm.log123);
        }

        [Test]
        public void TestInterfaceCalls()
        {
            NamingController sm = new NamingController();
            sm.call("snake_event", 1);
            sm.call("CamelEvent", 2);
            sm.call("event123", 3);
            sm.call("snake_event", 4);
            sm.call("CamelEvent", 5);
            sm.call("event123", 6);
            Assert.AreEqual(new List<int> { 1103, 1307, 1211 }, sm.finalLog);
            Assert.AreEqual(new List<int> { 1307 }, sm.snake_log);
            Assert.AreEqual(new List<int> { 1103 }, sm.CamelLog);
            Assert.AreEqual(new List<int> { 1211 }, sm.log123);
        }

    }
}

File: framec_tests/java/Naming/Naming_test.java
package framec_tests.java.Naming;

import java.util.*;
import static org.junit.Assert.assertEquals;
import org.junit.Test;

class NamingController extends Naming {

    NamingController() {
      super();
    }

protected void snake_action_do(int snake_param) {
    this.snake_log.add(snake_param);
}

protected void CamelAction_do(int CamelParam) {
    this.CamelLog.add(CamelParam);
}

protected void action123_do(int param123) {
    this.log123.add(param123);
}

protected void logFinal_do(int r) {
    this.finalLog.add(r);
}
}

public class Naming_test {
    /* Test that the generated state machine works and that events are
    named as expected. */
    @Test
    public void TestFollowingNamingWorks(){
        NamingController sm = new NamingController();
        sm.snake_event(1);
        assertEquals(sm.state_info(), "1");
        sm.snake_event(2);
        assertEquals(sm.state_info(), "0");
        sm.snake_event(1);
        assertEquals(sm.state_info(), "1");
        sm.CamelEvent(3);
        assertEquals(sm.state_info(), "0");
        sm.snake_event(1);
        assertEquals(sm.state_info(), "1");
        sm.event123(4);
        assertEquals(sm.state_info(), "0");
        ArrayList<Integer> expected = new ArrayList<Integer>();
        expected.add(1103);
        expected.add(1104);
        expected.add(1105);
        assertEquals(sm.finalLog, expected);
        sm.finalLog = new ArrayList<>();

        sm.CamelEvent(11);
        assertEquals(sm.state_info(), "2");
        sm.snake_event(2);
        assertEquals(sm.state_info(), "0");
        sm.CamelEvent(11);
        assertEquals(sm.state_info(), "2");
        sm.CamelEvent(3);
        assertEquals(sm.state_info(), "0");
        sm.CamelEvent(11);
        assertEquals(sm.state_info(), "2");
        sm.event123(4);
        assertEquals(sm.state_info(), "0");
        ArrayList<Integer> expected1 = new ArrayList<Integer>();
        expected1.add(1213);
        expected1.add(1214);
        expected1.add(1215);
        assertEquals(sm.finalLog, expected1);
        sm.finalLog = new ArrayList<>();

        sm.event123(21);
        assertEquals(sm.state_info(), "3");
        sm.snake_event(2);
        assertEquals(sm.state_info(), "0");
        sm.event123(21);
        assertEquals(sm.state_info(), "3");
        sm.CamelEvent(3);
        assertEquals(sm.state_info(), "0");
        sm.event123(21);
        assertEquals(sm.state_info(), "3");
        sm.event123(4);
        assertEquals(sm.state_info(), "0");
        ArrayList<Integer> expected2 = new ArrayList<Integer>();
        expected2.add(1323);
        expected2.add(1324);
        expected2.add(1325);
        assertEquals(sm.finalLog, expected2);
        ArrayList<Integer> expected3 = new ArrayList<Integer>();
        expected3.add(1103);
        expected3.add(1213);
        expected3.add(1323);
        assertEquals(sm.snake_log, expected3);
        ArrayList<Integer> expected4 = new ArrayList<Integer>();
        expected4.add(1104);
        expected4.add(1214);
        expected4.add(1324);
        assertEquals(sm.CamelLog, expected4);
        ArrayList<Integer> expected5 = new ArrayList<Integer>();
        expected5.add(1105);
        expected5.add(1215);
        expected5.add(1325);
        assertEquals(sm.log123, expected5);

    }
    /* Test that dynamic interface calls are renamed correctly. */
    @Test
    public void TestInterfaceCalls(){
        NamingController sm = new NamingController();

        sm.call("snake_event", 1);
        sm.call("CamelEvent", 2);
        sm.call("event123", 3);
        sm.call("snake_event", 4);
        sm.call("CamelEvent", 5);
        sm.call("event123", 6);

        ArrayList<Integer> expected = new ArrayList<Integer>();
        expected.add(1103);
        expected.add(1307);
        expected.add(1211);
        assertEquals(sm.finalLog, expected);
        ArrayList<Integer> expected1 = new ArrayList<Integer>();
        expected1.add(1307);
        assertEquals(sm.snake_log, expected1);
        ArrayList<Integer> expected2 = new ArrayList<Integer>();
        expected2.add(1103);
        assertEquals(sm.CamelLog, expected2);
        ArrayList<Integer> expected3 = new ArrayList<Integer>();
        expected3.add(1211);
        assertEquals(sm.log123, expected3);
    }
}

